### PR TITLE
feat: support ssr target compilation

### PR DESCRIPTION
**Pull request by wzc520pyfm**
- [x] ssr compiler
- [x] ssr compiler test

Now, vine can run on nuxt(ssr), ~~but I still need to write test cases for ssr compilation.~~

image

### CODE REVIEW

File: packages/compiler/src/index.ts [hunk 1]
```diff
@@ -96,6 +96,7 @@ export function compileVineTypeScriptFile(
   code: string,
   fileId: string,
   vineCompileCtx: VineCompileCtx,
+  ssr: boolean,
```
Comment (ShenQingchuan) [inline packages/compiler/src/index.ts:99]: Maybe set default value for `ssr = false` ? This allows us to omit passing false from those previous non-SSR test code

**Comment by ShenQingchuan** [comment]
What an excellent work! 💖 Thanks for @wzc520pyfm 
This is a big step!

**Comment by ShenQingchuan** [comment]
> What an excellent work! 💖 Thanks for @wzc520pyfm 
> 
> This is a big step!

Hi @wzc520pyfm 

Would you please share your Wechat ID so that I could invite you to our  core members' chat group

**Comment by wzc520pyfm** [comment]
> > What an excellent work! 💖 Thanks for @wzc520pyfm 
> > 
> > This is a big step!
> 
> Hi @wzc520pyfm 
> 
> Would you please share your Wechat ID so that I could invite you to our  core members' chat group

Of course! My Wechat ID is `wzc520pyf`.

File: packages/compiler/src/index.ts (hunk 1 above)
Comment (wzc520pyfm) [reply packages/compiler/src/index.ts:99]: > Maybe set default value for `ssr = false` ? This allows us to omit passing false from those previous non-SSR test code

Agree! Let me do it.

**Pull request merged**